id: PYSEC-0000-CVE-2021-29548
package:
  name: tensorflow
  ecosystem: PyPI
details: TensorFlow is an end-to-end open source platform for machine learning. An
  attacker can cause a runtime division by zero error and denial of service in `tf.raw_ops.QuantizedBatchNormWithGlobalNormalization`.
  This is because the implementation(https://github.com/tensorflow/tensorflow/blob/55a97caa9e99c7f37a0bbbeb414dc55553d3ae7f/tensorflow/core/kernels/quantized_batch_norm_op.cc)
  does not validate all constraints specified in the op's contract(https://www.tensorflow.org/api_docs/python/tf/raw_ops/QuantizedBatchNormWithGlobalNormalization).
  The fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit
  on TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as
  these are also affected and still in supported range.
affects:
  ranges:
  - type: GIT
    repo: https://github.com/tensorflow/tensorflow
    fixed: d6ed5bcfe1dcab9e85a4d39931bd18d99018e75b
  - type: ECOSYSTEM
    fixed: 2.1.4
  - type: ECOSYSTEM
    introduced: 2.2.0
    fixed: 2.2.3
  - type: ECOSYSTEM
    introduced: 2.3.0
    fixed: 2.3.3
  - type: ECOSYSTEM
    introduced: 2.4.0
    fixed: 2.4.2
references:
- type: WEB
  url: https://github.com/tensorflow/tensorflow/commit/d6ed5bcfe1dcab9e85a4d39931bd18d99018e75b
- type: WEB
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-p45v-v4pw-77jr
aliases:
- CVE-2021-29548
modified: "2021-05-20T15:17:00Z"
published: "2021-05-14T20:15:00Z"
